### DISCUSSION TITLE
Is it possible to get node-canvas working in Deta?

### DISCUSSION THREAD

**Discussion by SilentDepth** [deta/cloud-docs#442] [Help]
Working on a server-rendered image service which also has a browser-rendered version.

`<canvas>` makes things easy in browser. So I was looking for a canvas implementation for Node to reuse my codebase and I found node-canvas. But Deta seems don't support it...

I also tried PureImage but facing issues with font rendering.

Any help?

**Comment by BetaHuhn** [comment]
What issues are you facing exactly with [node-canvas](https://github.com/Automattic/node-canvas)?

Looking through their [issues](https://github.com/Automattic/node-canvas/issues?q=is:issue+lambda) it seems like the library has some problems running on AWS lambda which Deta Micros uses under the hood.

**Reply by SilentDepth** [reply]
Not able (or I don't know how) to view the actual logs. Visor shows nothing (even any access log). The endpoint just returns a 502 page from Deta itself.

I used to deploy the code to Vercel which also runs on AWS Lambda, and it works. However I need to manually set LD_LIBRARY_PATH env. Deta refuses to set this env. I'm not sure if it's the reason.
